style_guide.rst

```diff
@@ -0,0 +1,393 @@
+them, and are indented to the same level as that code.
+
+Inline Comments
+^^^^^^^^^^^^^^^
+
+Use inline comments sparingly.  An inline comment is a comment on the
+same line as a statement. Inline comments should be separated by at
+least two spaces from the statement.  They should start with a ``//`` and a
+single space.
+
+Inline comments are unnecessary and in fact distracting if they state
+the obvious. Don't do this:
+
+::
+
+   x = x + 1                 // Increment x
+
+But sometimes, this style of comment is useful:
+
+::
+
+   x = x + 1                 // Compensate for border
+
+
+
+Naming Conventions
+------------------
+
+Variable and function names should use the `snake_case <https://en.wikipedia.org/wiki/Snake_case>`_
+naming convention (i.e., ``lowercase_with_underscore``). For example:
+
+::
+
+          sum_of_squares
+          print_happy_birthday
+          total_apples
+
+
+Constants names should use snake_case with all caps:
+
+::
+
+    PI
+    MAX_CLIENTS
+    MAX_IRC_MSG_LEN
+
+
+Use descriptive names for parameter names, variables, and function
+names.  Use short names for local
+variables.  In general, the further away a variable will be used, the more
+descriptive the name needs to be.
+
+
+..
+    Printing logging / debug messages
+    ---------------------------------
+
+    All the projects in this class use a simple logging library called ``chilog`` that is documented in each of the project specifications. You must use the ``chilog`` functions *exclusively* for printing logging or debug messages. **Do not use printf() directly in your code**. Please note that the ``chilog`` functions provide essentially the same functionality as ``printf``, so there is no situation where ``printf`` would be necessary instead of ``chilog`` (using ``chilog`` consistently also means you will not have to scrub ``printf``'s from your code before submitting it).
+
+    Furthermore, all the messages at the ``INFO``, ``WARNING``, ``ERROR``, and ``CRITICAL`` levels must be used only for their intended purposes (e.g., only use ``ERROR`` to print out actual errors in the execution of your program). You must use the ``DEBUG`` level only to print informative debug messages that would be understood by any developer trying to debug your code. You may use the ``TRACE`` level to print *any* debug message (including those that would only be understood by you). However, if your code is riddled with ``TRACE`` logging statements (including commented out ones) to the point where it is hard to read the code itself, we may take points off for this. So, once a ``TRACE`` logging statement has served its purpose, we suggest you remove it (not just comment it out).
+
+    You should assume that graders will run your code with logging at the ``INFO`` level, and will only use the ``DEBUG`` level if they need to debug an issue with your code. We will never run your code with logging at the ``TRACE`` level.
```

Comment:
This also doesn't really apply given our current grading scheme, right?

Reply:
See above (this section is commented out)